src/ocm/components/newFeatureSupportLevels/NewFeatureSupportLevelProvider.tsx

```diff
@@ -0,0 +1,140 @@
+import React, { PropsWithChildren } from 'react';
+import {
+  ActiveFeatureConfiguration,
+  Cluster,
+  CpuArchitecture,
+  FeatureId,
+  getDefaultCpuArchitecture,
+  SupportLevel,
+  SupportLevels,
+} from '../../../common';
+import { useOpenshiftVersions } from '../../hooks';
+import { getNewFeatureDisabledReason, isFeatureSupported } from './newFeatureStateUtils';
+import useInfraEnv from '../../hooks/useInfraEnv';
+import {
+  NewFeatureSupportLevelContextProvider,
+  NewFeatureSupportLevelData,
+  NewFeatureSupportLevelMap,
+} from '../../../common/components/newFeatureSupportLevels';
+import useSupportLevelsAPI from '../../hooks/useSupportLevelsAPI';
+import { TFunction } from 'i18next';
+
+export type NewSupportLevelProviderProps = PropsWithChildren<{
+  clusterFeatureUsage?: string;
+  openshiftVersion?: string;
+  loadingUi: React.ReactNode;
+  cluster?: Cluster;
+  cpuArchitecture?: string;
+}>;
+
+export const getFeatureSupported = (featureSupportLevels: SupportLevels, featureId: FeatureId) => {
+  return featureSupportLevels && featureSupportLevels[featureId] !== 'unsupported';
+};
+
+export const NewFeatureSupportLevelProvider: React.FC<NewSupportLevelProviderProps> = ({
+  cluster,
+  children,
+  loadingUi,
+  cpuArchitecture,
+  openshiftVersion,
+}) => {
+  const { loading: loadingOCPVersions } = useOpenshiftVersions();
+  const { infraEnv, isLoading: isInfraEnvLoading } = useInfraEnv(
+    cluster?.id || '',
+    CpuArchitecture.USE_DAY1_ARCHITECTURE,
+  );
+  const featureSupportLevels = useSupportLevelsAPI('features', openshiftVersion, cpuArchitecture);
+
+  const supportLevelData = React.useMemo<NewFeatureSupportLevelMap>(() => {
+    if (!featureSupportLevels) {
+      return {} as NewFeatureSupportLevelMap;
+    }
+    return featureSupportLevels;
+  }, [featureSupportLevels]);
+
+  const isLoading = !supportLevelData || loadingOCPVersions || isInfraEnvLoading;
+
+  const activeFeatureConfiguration = React.useMemo<ActiveFeatureConfiguration>(
+    () => ({
+      underlyingCpuArchitecture: (infraEnv?.cpuArchitecture ||
+        cluster?.cpuArchitecture ||
+        getDefaultCpuArchitecture()) as CpuArchitecture,
```

Comment:
This assertion is wrong. Neither the infraEnv's nor the cluster's CPU architecture ar of time CpuArchitecture, the former are strings the later is an enum